src/engine/source/hlp/src/SpecificParsers.cpp

```diff
@@ -376,3 +378,113 @@ bool parseURL(const char **it, char endToken, URLResult &result) {
 
     return true;
 }
+
+bool parseDomain(const char **it, char endToken, std::vector<std::string> const& captureOpts, DomainResult &result) {
+    static const std::string_view valid_chars {"abcdefghijklmnopqrstuvwxyz"
+                                               "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+                                               "0123456789-_."};
+    static const int DOMAIN_MAX_SIZE = 253;
+    static const int LABEL_MAX_SIZE = 63;
+
+    const char *start = *it;
+    while (**it != endToken || **it != '\0') {
+        (*it)++;
+    }
+    std::string_view str { start, (size_t)((*it) - start) };
+
+    size_t protocol_end = str.find("://");
+    size_t domain_start = 0;
+    // Protocol
+    if (std::string::npos != protocol_end) {
+        result.protocol = str.substr(0, protocol_end);
+        domain_start = protocol_end + 3;
+    }
+    size_t domain_end = str.find("/", protocol_end + 3);
+    // Domain
+    result.domain = str.substr(domain_start, domain_end - domain_start);
+    // Domain length check
+    if (result.domain.empty() || result.domain.length() > DOMAIN_MAX_SIZE){
+        //TODO Log domain size error
+        return false;
+    }
+    // Domain valid characters check
+    for (char const &c: result.domain) {
+        if (std::string::npos == valid_chars.find(c)) {
+            //TODO Log invalid char error
+            return false;
+        }
+    }
+    // Route
+    if (std::string::npos != domain_end) {
+        result.route = str.substr(domain_end+1);
+    }
+
+    std::vector<std::string> labels;
```

Comment:
could be a` std::vector<std::string_view>` to avoid the copies

Reply:
The labels are being deleted once they are parsed into a valid field because the parser logic goes backward when searching for TLD, but appends the labels upstream when forming each result.
Changing this to string_view requires modifying this logic to conserve the labels until the end, like using iterators to identify where was the TLD label, etc.
From the call we have: This shouldn´t be as expensive to generate a bottleneck.
A comment will be added, and this string usage will be kept until it´s identified that a change is required.